tests/rados/customer_scenarios.py

```diff
@@ -0,0 +1,451 @@
+    for val in osd_nodes:
+        for osd in osd_daemons:
+            if re.search(osd["hostname"], val.hostname):
+                # Not using the container ID's provided in ceph orch ps command.
+                # Bug : https://bugzilla.redhat.com/show_bug.cgi?id=1943494
+                # cmd = f"podman restart {osd['container_id']}"
+                cmd = f"systemctl restart ceph-{cluster_fsid}@osd.{osd['daemon_id']}.service"
+                log.info(
+                    f"rebooting osd-{osd['daemon_id']} on host {osd['hostname']}. Command {cmd}"
+                )
+                val.exec_command(sudo=True, cmd=cmd)
+                # Sleeping for 5 seconds for status to be updated
+                time.sleep(5)
+                break
+
+    # Re-weighting the OSd's based on usage to trigger rebalance
+    # todo: Verify re-balancing process on OSD's ( PG movement across cluster)
+    # todo: Add re-balancing based on crush item provided
+    # BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1766702
+    reweight_crush_items(node=node)
+
+    """
+    Waiting for 2 hours for cluster to get to active + clean state.
+    Rationale: during cluster activities like backfill, rebalance, osd map change cause Mon DB to be updated.
+    Hence, we can wait till mon DB updates are completed, after which DB size should be reduced,
+    by trimming the old mappings once the new mappings are added.
+    If cluster healthy state is reached within 2 hours, we exit the loop earlier, without waiting for stipulated time,
+    But if cluster is still performing operations for long time,
+     we would need at-least some data to make sure DB is not just increasing.
+      ( DB is expected to increase when operations are in progress.
+       Old mappings are removed when operations/ new updates are completed. )
+    """
+    end_time = datetime.datetime.now() + datetime.timedelta(seconds=7200)
+    while end_time > datetime.datetime.now():
+        status_report = run_ceph_command(node, action="ceph_report")
+        ceph_health_status = status_report["health"]
+        recovery_tuple = ("OSD_DOWN", "PG_AVAILABILITY", "PG_DEGRADED", "SLOW_OPS")
+        daemon_info = run_ceph_command(node, action="daemon_info")
+        mon_daemons = [entry for entry in daemon_info if entry["daemon_type"] == "mon"]
+        mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+
+        # Checking for any health warnings that increase db size
+        flag = (
+            True
+            if not any(
+                key in ceph_health_status["checks"].keys() for key in recovery_tuple
+            )
+            else False
+        )
+
+        # Proceeding to check if all PG's are in active + clean
+        if flag:
+            for entry in status_report["num_pg_by_state"]:
+                rec = ("remapped", "backfilling", "degraded")
+                if any(key in rec for key in entry["state"].split("+")):
```

Comment:
Would you be able to give the value of `entry['state']`? List comprehension is always a costly operation... we could do things quicker if we exit as soon as we hit a match instead of gathering/traversing the complete list.

Reply:
Split returns a list and we are looking for any one of the rec items... this is good.